compiler/rustc_errors/src/diagnostic.rs

```diff
@@ -243,12 +243,15 @@ impl Diagnostic {
 
     pub fn is_error(&self) -> bool {
         match self.level {
-            Level::Bug | Level::DelayedBug | Level::Fatal | Level::Error | Level::FailureNote => {
-                true
-            }
+            Level::Bug
+            | Level::DelayedBug(DelayedBugKind::Normal)
+            | Level::Fatal
+            | Level::Error
+            | Level::FailureNote => true,
 
             Level::ForceWarning(_)
             | Level::Warning
+            | Level::DelayedBug(DelayedBugKind::GoodPath)
```

Comment:
This seems wrong. Good path delayed bugs, when emitted, are errors just as much as normal delayed bugs.

Reply:
No. Good path bugs are not errors, since they are defused by emitting warnings *or* error diagnostics.

Returning an ErrorGuaranteed here would be promising that compilation *fails*.

Reply:
I think you're looking at the wrong end. If you fail to emit a warning or an error, the the good path delayed bug is emitted as a bug. It has "bug" in the name!